### PR TITLE
Show proper connector label in now playing notifications

### LINKED ISSUE #1020: Wrong connector title in "now playing" notifications
Every connector placed inside `connectors.js` after:

```
        {
            label: 'Sonerezh',
            matches: ['*://sonerezh.*/*', '*://*/*sonerezh*'],
            js: ['connectors/v2/sonerezh.js'],
            version: 2
        }
```

is shown as
screenshot_8

Looks like this part inside `song.js` particularly likes `['*://sonerezh.*/*', '*://*/*sonerezh*']`
screenshot_9

### PR DESCRIPTION
**Pull request by alexesprit**
Resolves #1020.

### CODE REVIEW

File: core/background/objects/song.js [hunk 1]
```diff
@@ -50,15 +50,14 @@ define([
 		/**
 		 * Various optional data
 		 */
-		var connector = _.find(connectors, function(connector) {
-			return _.find(connector.matches, function(patternStr) {
-				var pattern = RoutePattern.fromString(patternStr);
```
Comment (inverse) [inline core/background/objects/song.js]: What exactly was this RoutePattern lib doing?

Comment (alexesprit) [inline core/background/objects/song.js]: It was searching for matches between some URL match pattern and some URL.
The code in `inject.js` does it better.

**Comment by alexesprit** [comment]
I found a way to fix the issue w/o using RoutePattern/URL mather:
```diff
diff --git a/core/background/controller.js b/core/background/controller.js
index eabd7f9..8e2a577 100644
--- a/core/background/controller.js
+++ b/core/background/controller.js
@@ -95,7 +95,8 @@ define([
 				// clear any previous song and its bindings
 				resetState();
 
-				currentSong = new Song(newState);
+				currentSong = new Song(newState, connector);
+
 				bindSongListeners(currentSong);
 
 				console.log('Tab ' + tabId + ': new ' + (isReplayingSong ? '(replaying) ' : '') + 'song detected: ' + JSON.stringify(currentSong.attr()));
diff --git a/core/background/objects/song.js b/core/background/objects/song.js
index 2ea81ea..37e2a05 100644
--- a/core/background/objects/song.js
+++ b/core/background/objects/song.js
@@ -4,15 +4,12 @@
  * Song object
  */
 define([
-	'wrappers/can',
-	'connectors',
-	'vendor/route-pattern',
-	'underscore'
-], function(can, connectors, RoutePattern, _) {
+	'wrappers/can'
+], function(can) {
 	/**
 	 * @constructor
 	 */
-	return function(parsedData) {
+	return function(parsedData, connector) {
 		/**
 		 * Number of seconds of playback before the track is scrobbled.
 		 * This value is used only if no duration was parsed or loaded
@@ -50,16 +47,6 @@ define([
 		/**
 		 * Various optional data
 		 */
-		var connector = _.find(connectors, function(connector) {
-			return _.find(connector.matches, function(patternStr) {
-				var pattern = RoutePattern.fromString(patternStr);
-				if (parsed.url !== null) {
-					return pattern.matches(parsed.url.href);
-				}
-				return false;
-			});
-		}) || null;
-
 		var metadata = {
 			userloved: parsedData.userloved === 1,
 			startTimestamp: Math.floor(Date.now() / 1000), // UTC timestamp in seconds
```
Is this more acceptable?

**Comment by david-sabata** [comment]
@alexesprit that looks way better :wink:

**Comment by alexesprit** [comment]
Updated.

**Pull request merged**